expression/simple_rewriter.go

```diff
@@ -58,7 +58,7 @@ func ParseSimpleExprCastWithTableInfo(ctx sessionctx.Context, exprStr string, ta
 	if err != nil {
 		return nil, err
 	}
-	e = BuildCastFunction(ctx, e, targetFt)
+	e = BuildCastFunction(ctx, e, targetFt.Clone())
```

Comment:
Why change this?

Reply:
From #19029, I think it's reasonable not to let column and scalarFunction share the same ft. Though it's not necessary for this PR.